config/initializers/admin.rb

```diff
@@ -0,0 +1,7 @@
+require_relative "../../lib/restrict_admin_by_ip_middleware"
+
+Rails.application.configure do
+  allowed_ips = ENV.fetch("ADMIN_ALLOWED_IPS", "").split(",")
+
+  config.middleware.use RestrictAdminByIpMiddleware, allowed_ips
```

Comment:
Should this middleware be inserted lower in the stack to minimise the amount of app/framework code the request will execute?

Reply:
I'm not totally clear on the stack. Is it ok to put it at position 0?

Reply:
`rails middleware` will output the current stack, and the internal Rails stack is documented here: https://guides.rubyonrails.org/rails_on_rack.html#internal-middleware-stack

Perhaps after `Rails::Rack::Logger` so the request will be included in the logs?